js_modules/dagster-ui/packages/ui-core/src/app/__tests__/BaseFallthroughRoot.test.tsx

```diff
@@ -0,0 +1,103 @@
+import {MockedProvider} from '@apollo/client/testing';
+import {render, waitFor} from '@testing-library/react';
+import {MemoryRouter, Route, Switch} from 'react-router';
+
+import {__ANONYMOUS_ASSET_JOB_PREFIX} from '../../asset-graph/Utils';
+import {
+  buildAssetGroup,
+  buildPipeline,
+  buildRepository,
+  buildRepositoryLocation,
+  buildWorkspaceLocationEntry,
+} from '../../graphql/types';
+import {testId} from '../../testing/testId';
+import {WorkspaceProvider} from '../../workspace/WorkspaceContext/WorkspaceContext';
+import {buildWorkspaceMocks} from '../../workspace/WorkspaceContext/__fixtures__/Workspace.fixtures';
+import {BaseFallthroughRoot} from '../BaseFallthroughRoot';
+import {
+  workspaceWithDunderJob,
+  workspaceWithJob,
+  workspaceWithNoRepos,
+} from '../__fixtures__/useJobStateForNav.fixtures';
+
+describe('BaseFallthroughRoot', () => {
+  const Test = () => (
+    <MemoryRouter initialEntries={[{pathname: '/'}]}>
+      <WorkspaceProvider>
+        <Switch>
+          <Route path="/overview" render={() => <div />} />
+          <Route path="/locations" render={() => <div />} />
+          <BaseFallthroughRoot />
+        </Switch>
+        <Route render={(m) => <div data-testid={testId('path')}>{m.location.pathname}</div>} />
+      </WorkspaceProvider>
+    </MemoryRouter>
+  );
+
+  it('redirects to /locations if there are no repositories', async () => {
+    const {getByTestId} = render(
+      <MockedProvider mocks={[...workspaceWithNoRepos]}>
+        <Test />
+      </MockedProvider>,
+    );
+    await waitFor(() => {
+      expect(getByTestId('path').textContent).toEqual('/locations');
+    });
+  });
+
+  it('redirects to the asset graph if assetGroups are present and repos have no visible jobs', async () => {
+    const {getByTestId} = render(
+      <MockedProvider
+        mocks={buildWorkspaceMocks([
+          buildWorkspaceLocationEntry({
+            name: 'some_workspace',
+            locationOrLoadError: buildRepositoryLocation({
+              name: 'location_with_dunder_job',
+              repositories: [
+                buildRepository({
+                  name: `repo_with_pseudo_job`,
+                  assetGroups: [buildAssetGroup()],
```

Comment:
Probably worth giving this a name, since I think there's the possibility of the randomly generated string in the mock builder changing in the future.

Reply:
Oh good call will fix this